scripts/provider_watch.py

```
@@ -0,0 +1,1525 @@
+            body = json.dumps(json_body, separators=(",", ":")).encode("utf-8")
+            request_headers["Content-Type"] = "application/json"
+
+        last_error: Exception | None = None
+        for attempt in range(self.retries + 1):
+            try:
+                request = Request(
+                    url,
+                    data=body,
+                    headers=request_headers,
+                    method=method,
+                )
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)